packages/deployments/contracts/contracts/core/connext/facets/InboxFacet.sol

```diff
@@ -235,15 +235,24 @@ contract InboxFacet is BaseConnextFacet {
   ) internal returns (address, uint256) {
     bytes32 _canonicalId = _tokenId.id();
     uint32 _canonicalDomain = _tokenId.domain();
+
+    // Load amount once.
+    uint256 _amount = _action.amnt();
+
+    // Check for the empty case -- if it is 0 value there is no strict requirement the
+    // canonical information be defined (i.e. you can supply address(0) to xcall). If this
+    // is the case, return _token as address(0)
+    if (_amount == 0 && _canonicalDomain == 0 && _canonicalId == bytes32(0)) {
```

Comment:
Shouldn't this have an `emit Receive(...)`?  Because the other 0 case (line 256) also has an emit.